presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -384,29 +384,33 @@ static boolean shouldUseRecordReaderFromInputFormat(Configuration configuration,
                 .anyMatch(USE_RECORD_READER_FROM_INPUT_FORMAT_ANNOTATION::equals);
     }
 
-    static boolean shouldUseFileSplitsFromInputFormat(InputFormat<?, ?> inputFormat, Configuration conf, String tablePath)
+    static boolean shouldUseFileSplitsFromInputFormat(InputFormat<?, ?> inputFormat, DirectoryLister directoryLister)
     {
-        boolean hasUseSplitsAnnotation = Arrays.stream(inputFormat.getClass().getAnnotations())
-                .map(Annotation::annotationType)
-                .map(Class::getSimpleName)
-                .anyMatch(USE_FILE_SPLITS_FROM_INPUT_FORMAT_ANNOTATION::equals);
+        if (directoryLister instanceof HudiDirectoryLister) {
+            boolean hasUseSplitsAnnotation = Arrays.stream(inputFormat.getClass().getAnnotations())
+                    .map(Annotation::annotationType)
+                    .map(Class::getSimpleName)
+                    .anyMatch(USE_FILE_SPLITS_FROM_INPUT_FORMAT_ANNOTATION::equals);
+
+            return hasUseSplitsAnnotation &&
+                    (!isHudiParquetInputFormat(inputFormat) || shouldUseFileSplitsForHudi(inputFormat, ((HudiDirectoryLister) directoryLister).getMetaClient()));
+        }
 
-        return hasUseSplitsAnnotation && (!isHudiParquetInputFormat(inputFormat) || shouldUseFileSplitsForHudi(inputFormat, conf, tablePath));
+        return false;
     }
 
     static boolean isHudiParquetInputFormat(InputFormat<?, ?> inputFormat)
     {
         return inputFormat instanceof HoodieParquetInputFormat;
     }
 
-    private static boolean shouldUseFileSplitsForHudi(InputFormat<?, ?> inputFormat, Configuration conf, String tablePath)
+    private static boolean shouldUseFileSplitsForHudi(InputFormat<?, ?> inputFormat, Optional<HoodieTableMetaClient> metaClient)
     {
         if (inputFormat instanceof HoodieParquetRealtimeInputFormat) {
```

Comment:
is n't this check same as `!isHudiParquetInputFormat(inputFormat)` above? any chances to simplify?

Reply:
`HoodieParquetRealtimeInputFormat` is subclass of `HoodieParquetInputFormat`. However, we want to use file splits from the former but not latter. Hence, a separate check.